src/rail_gui.cpp

```diff
@@ -169,13 +173,20 @@ static void PlaceRail_Waypoint(TileIndex tile)
 	}
 }
 
+/**
+ * Helper macro to return the StationClassID of a given StationClass instance. This is needed in order
+ * to provide the StationClassID to specific methods that require it like to draw the sprites and send
+ * the build station command over the network.
+ */
+#define GET_STATION_CLASS_ID(X) StationClass::Allocate(X->global_id)
```

Comment:
This feels like the wrong way round. Why not continue to use StationClassID and use StationClass::Get where needed? Seems like fewer changes would be needed overall

Reply:
Hi LordAro and thanks for your quick review!
I am flipping things around and yep you are right! For some reason I though I needed to hold a pointer to the object in GUIStationClassList. I am almost done pushing an update, will resolve the comment once all works fine.